hadoop-hdds/test-utils/src/main/java/org/apache/hadoop/test/JacocoServer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.test;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import org.jacoco.core.data.ExecutionDataWriter;
+import org.jacoco.core.runtime.RemoteControlReader;
+import org.jacoco.core.runtime.RemoteControlWriter;
+
+/**
+ * Simple TPC server to collect all the Jacoco coverage data.
+ */
+public final class JacocoServer {
+
+  private static int port = 6300;
+
+  private static String destinationFile = "/tmp/jacoco-combined.exec";
+
+  private JacocoServer() {
+  }
+
+  @SuppressWarnings("checkstyle:EmptyStatement")
+  public static void main(String[] args) throws IOException {
+    ExecutionDataWriter destination =
+        new ExecutionDataWriter(new FileOutputStream(destinationFile));
+    ServerSocket serverSocket = new ServerSocket(port);
+    Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+      try {
+        destination.flush();
+        serverSocket.close();
+      } catch (Exception ex) {
+        ex.printStackTrace();
+      }
+    }));
+
+    while (true) {
+      final Socket socket = serverSocket.accept();
+      new Thread(() -> {
+        try {
+          RemoteControlWriter writer =
+              new RemoteControlWriter(socket.getOutputStream());
+          RemoteControlReader reader =
+              new RemoteControlReader(socket.getInputStream());
+          reader.setSessionInfoVisitor(destination::visitSessionInfo);
+          reader.setExecutionDataVisitor(destination::visitClassExecution);
+          while (reader.read()) {
+            ;//read until the end of the stream.
+          }
+          destination.flush();
```

Comment:
I've found that these calls to `ExecutionDataWriter`'s methods should be [synchronized](https://github.com/adoroszlai/hadoop-ozone/commit/7d21d2c5f7147e879397453f57e3d8d67e91cc99) to avoid producing garbage files:

```
Exception in thread "main" java.io.UTFDataFormatException: malformed input around byte 2
  at java.io.DataInputStream.readUTF(DataInputStream.java:634)
  at java.io.DataInputStream.readUTF(DataInputStream.java:564)
  at org.jacoco.cli.internal.core.data.ExecutionDataReader.readExecutionData(ExecutionDataReader.java:149)
```

or similar `Unknown block type` error.

Reply:
Thanks to investigate it. I uploaded a synchronized version. 

The main problem is that the same functionality is written in Jacoco project under LGPL which couldn't be imported. I wrote my own version from scratch, and this mistake clearly proves that it's independent, as I added my own mistakes ;-)

Reply:
`destination.flush();` should be synced, too.

Reply:
added.